.dotnet/src/Generated/Models/InternalCreateModerationResponseResultCategoryAppliedInputTypes.cs

```diff
@@ -0,0 +1,81 @@
+        }
+
+        internal InternalCreateModerationResponseResultCategoryAppliedInputTypes(IReadOnlyList<string> hate, IReadOnlyList<string> hateThreatening, IReadOnlyList<string> harassment, IReadOnlyList<string> harassmentThreatening, IReadOnlyList<string> illicit, IReadOnlyList<string> illicitViolent, IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSelfHarm1> selfHarm, IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSelfHarmIntent> selfHarmIntent, IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSelfHarmInstruction> selfHarmInstructions, IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSexual> sexual, IReadOnlyList<string> sexualMinors, IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesViolence> violence, IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesViolenceGraphic> violenceGraphic, IDictionary<string, BinaryData> serializedAdditionalRawData)
+        {
+            Hate = hate;
+            HateThreatening = hateThreatening;
+            Harassment = harassment;
+            HarassmentThreatening = harassmentThreatening;
+            Illicit = illicit;
+            IllicitViolent = illicitViolent;
+            SelfHarm = selfHarm;
+            SelfHarmIntent = selfHarmIntent;
+            SelfHarmInstructions = selfHarmInstructions;
+            Sexual = sexual;
+            SexualMinors = sexualMinors;
+            Violence = violence;
+            ViolenceGraphic = violenceGraphic;
+            SerializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        internal InternalCreateModerationResponseResultCategoryAppliedInputTypes()
+        {
+        }
+
+        public IReadOnlyList<string> Hate { get; }
+        public IReadOnlyList<string> HateThreatening { get; }
+        public IReadOnlyList<string> Harassment { get; }
+        public IReadOnlyList<string> HarassmentThreatening { get; }
+        public IReadOnlyList<string> Illicit { get; }
+        public IReadOnlyList<string> IllicitViolent { get; }
+        public IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSelfHarm1> SelfHarm { get; }
+        public IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSelfHarmIntent> SelfHarmIntent { get; }
+        public IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSelfHarmInstruction> SelfHarmInstructions { get; }
+        public IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesSexual> Sexual { get; }
+        public IReadOnlyList<string> SexualMinors { get; }
+        public IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesViolence> Violence { get; }
+        public IReadOnlyList<InternalCreateModerationResponseResultCategoryAppliedInputTypesViolenceGraphic> ViolenceGraphic { get; }
```

Comment:
nit: I was expecting all of these to follow the same pattern. Specifically, I would have guessed that all of them would be IReadOnlyList properties of some enum. The enum would have one ("text") or two ("text" or "image") possible value.

Could this be suggesting that there is something wrong with the TypeSpec in the case where there is only one possible value?

Reply:
It looks that way, yes.

Comparing `Illict` (adopts `<string>`) and `SelfHarm (adopts `<enum>`):

```javascript
      /** The applied input type(s) for the category 'illicit'. */
      illicit: "text"[];
      /** The applied input type(s) for the category 'self-harm'. */
      `self-harm`: ("text" | "image")[];
```

Things that massage this:

- Changing `illicit` to `("text" | "text")[]` achieves the intended generation outcome of getting a type, but (a) it looks very weird; and (b) it emits non-compiling code (an enum with the value `Text` appearing twice)
- Changing it to `(string | "text")[]` makes it behave exactly as intended in .NET, but the origin spec really doesn't make this out to be a true extensible enum (and we don't treat it as such)

Reply:
Looking at this... sigh... time to break out `options.yaml` again. We can make this clean (and better than the source) if we customize and formalize the commonality of the input types.

Reply:
This still ends up emitting suboptimally without customization. I'm going to keep the customization because it's very nice for the spec, but there's likely some feature request fodder lurking here to make the emitter cleaner.

With `union ModerationAppliedInputType { string, text: "text", image: "image" }`:

- `foo: ModerationAppliedInputType.text[]` emits as an `IReadOnlyList<string>`
- `foo: (ModerationAppliedInputType.text | ModerationAppliedInputType.image)[]` emits as `IReadOnlyList<BinaryData>`
- `foo: ModerationAppliedInputType[]` emits as `IReadOnlyList<ModerationAppliedInputType>`

The last one is the nicest, but requires losing constraint information in the spec -- which we really don't want.